### PR TITLE
mes-3673: restructuring folders for category B non-pass-finalisation …

### PR DESCRIPTION
**Pull request by MarkPatt89**
…and introducing folder structure for category BE

## Description
- Introducing category B and category BE file structure for `non-pass-finalisation` pages
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/non-pass-finalisation/cat-b/__tests__/non-pass-finalisation.cat-b.page.spec.ts [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
+import { ActivityCodeComponent } from '../../../office/components/activity-code/activity-code';
+import { SetTestStatusWriteUp } from '../../../../modules/tests/test-status/test-status.actions';
+import * as testActions from '../../../../modules/tests/tests.actions';
 import { TestFinalisationComponentsModule } from
-'../../../components/test-finalisation/test-finalisation-component.module';
-import { TestCategory } from '../../../shared/models/test-category';
+'../../../../components/test-finalisation/test-finalisation-component.module';
+import { TestCategory } from '../../../../shared/models/test-category';
 
 describe('NonPassFinalisationPage', () => {
```
Comment (TomBillingtonUK) [inline src/pages/non-pass-finalisation/cat-b/__tests__/non-pass-finalisation.cat-b.page.spec.ts]: Can you update the describe string to include cat b

File: src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { NonPassFinalisationCatBEPage } from '../non-pass-finalisation.cat-be.page';
+import { NonPassFinalisationViewDidEnter } from '../../non-pass-finalisation.actions';
+import { ActivityCodeComponent } from '../../../office/components/activity-code/activity-code';
+import { SetTestStatusWriteUp } from '../../../../modules/tests/test-status/test-status.actions';
+import * as testActions from '../../../../modules/tests/tests.actions';
+import { TestFinalisationComponentsModule } from
+  '../../../../components/test-finalisation/test-finalisation-component.module';
+import { TestCategory } from '../../../../shared/models/test-category';
+
+describe('NonPassFinalisationPage', () => {
```
Comment (TomBillingtonUK) [inline src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts]: Could you update this one to to BE

File: src/pages/non-pass-finalisation/cat-b/__tests__/non-pass-finalisation.cat-b.page.spec.ts (hunk 1 above)
Comment (MarkPatt89) [reply src/pages/non-pass-finalisation/cat-b/__tests__/non-pass-finalisation.cat-b.page.spec.ts]: Yes, thanks for catching this.

File: src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts (hunk 2 above)
Comment (MarkPatt89) [reply src/pages/non-pass-finalisation/cat-be/__tests__/non-pass-finalisation.cat-be.page.spec.ts]: And this... lol

File: src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  d255$: Observable<boolean>;
+  isWelshTest$: Observable<boolean>;
+}
+
+@IonicPage()
+@Component({
+  selector: 'non-pass-finalisation-cat-be-page',
+  templateUrl: 'non-pass-finalisation.cat-be.page.html',
+})
+export class NonPassFinalisationCatBEPage extends PracticeableBasePageComponent {
```
Comment (TomBillingtonUK) [inline src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.ts]: On B+E Pages we want to change the page to extend the BasePageComponent as there is no Practice Mode.

File: src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  ionViewDidEnter(): void {
+    this.store$.dispatch(new NonPassFinalisationViewDidEnter());
+  }
+
+  continue() {
+    Object.keys(this.form.controls).forEach(controlName => this.form.controls[controlName].markAsDirty());
+    if (this.form.valid) {
+      this.store$.dispatch(new SetTestStatusWriteUp(this.slotId));
+      this.store$.dispatch(new PersistTests());
+      this.navController.push(CAT_B.BACK_TO_OFFICE_PAGE);
```
Comment (TomBillingtonUK) [inline src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.ts]: Incorrect navigation

File: src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
+<ion-header>
+  <practice-mode-banner *ngIf="isPracticeMode"></practice-mode-banner>
```
Comment (TomBillingtonUK) [inline src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.html]: can be removed when base page is updated

File: src/pages/post-debrief-holding/post-debrief-holding.html [hunk 6]
```diff
@@ -6,9 +6,9 @@
     <img id='dvsa-logo' src='../../assets/imgs/back-to-office/logo.png' />
   </div>
   <div class="container">
-    <button ion-button class="mes-neutral-button office-button top-button" navPush="NonPassFinalisationPage"
+    <button ion-button class="mes-neutral-button office-button top-button" navPush="NonPassFinalisationCatBPage"
```
Comment (TomBillingtonUK) [inline src/pages/post-debrief-holding/post-debrief-holding.html:9]: yuck this should be using the enums, maybe extract to a function if we can't do it in the template

**Comment by TomBillingtonUK** [comment]
You are also missing a module file bringing all the page modules together like waiting-room-to-car.module.ts

File: src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.ts (hunk 4 above)
Comment (MarkPatt89) [reply src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.ts]: Done

File: src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.html (hunk 5 above)
Comment (MarkPatt89) [reply src/pages/non-pass-finalisation/cat-be/non-pass-finalisation.cat-be.page.html]: Done

**Review by TomBillingtonUK — APPROVED** [review]
Looks Good

**Review by appleabs — APPROVED** [review]

**Review by rossfauldsbjss — APPROVED** [review]
looks good

**Pull request merged**